app/controllers/planning_applications/assessment/local_policy_areas_controller.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+module PlanningApplications
+  module Assessment
+    class LocalPolicyAreasController < AuthenticationController
+      before_action :set_planning_application
+      before_action :set_local_policy, only: %i[create destroy edit new show update]
+      before_action :set_local_policy_area, only: %i[edit show update]
+
+      def show
+        respond_to do |format|
+          format.html
+        end
+      end
+
+      def new
+        @local_policy_area = @local_policy.local_policy_areas.new
+        respond_to do |format|
+          format.html
+        end
+      end
+
+      def edit
+        respond_to do |format|
+          format.html
+        end
+      end
+
+      def destroy
+        @local_policy.local_policy_areas.destroy(params[:id])
+
+        redirect_to new_planning_application_assessment_local_policy_path(@planning_application)
+      end
+
+      def create
+        @local_policy_area = @local_policy.local_policy_areas.new(local_policy_params.except(:policy))
+
+        if @local_policy_area.save
+          redirect_to edit_planning_application_assessment_local_policy_path(@planning_application, @planning_application.local_policy)
+        else
+          respond_to do |format|
+            format.html { render :new }
+          end
+        end
+      end
+
+      def update
+        if @local_policy_area.update(local_policy_params)
+          redirect_to edit_planning_application_assessment_local_policy_path(@planning_application, @planning_application.local_policy),
+            notice: I18n.t("local_policies.successfully_updated")
+        else
+          respond_to do |format|
+            format.html { render :new }
+          end
+        end
+      end
+
+      private
+
+      def set_local_policy
+        @local_policy = @planning_application.local_policy || @planning_application.create_local_policy!
+      end
+
+      def set_local_policy_area
+        @local_policy_area = @local_policy.local_policy_areas.find(params[:id])
+      end
+
+      def local_policy_params
+        params.require(:local_policy_area).permit([:area, :id, :policies, :policy, :assessment, :guidance, :local_policy_id])
```

Comment:
If we always scope our actions through the association then we don't need the `:local_policy_id` in the params which would allow someone to detach a record from one policy and attach it to another.

Same goes for `:id` as well - shouldn't be permitted here.